src/net/parse.cpp

```diff
@@ -34,28 +34,69 @@
 
 namespace net
 {
+    void get_network_address_host_and_port(const std::string& address, std::string& host, std::string& port)
+    {
+        // require ipv6 address format "[addr:addr:addr:...:addr]:port"
+        if (address.find(']') != std::string::npos)
+        {
+            host = address.substr(1, address.rfind(']') - 1);
+            if ((host.size() + 2) < address.size())
+            {
+                port = address.substr(address.rfind(':') + 1);
+            }
+        }
+        else
+        {
+            host = address.substr(0, address.rfind(':'));
+            if (host.size() < address.size())
+            {
+                port = address.substr(host.size() + 1);
+            }
```

Comment:
This function can also use boost from_string

Reply:
That function takes an address that may be specified as "address:port" and separates the two; address_v6.from_string and address_v4.from_string -- ok, they're deprecated, so make_address_vN() -- use inet_pton under the hood which would not play nice with the ":port" part nor the square braces around the address in the case of ipv6.